modules/service/src/main/scala/ch/epfl/bluebrain/nexus/commons/service/persistence/SequentialTagIndexer.scala

```diff
@@ -43,58 +51,69 @@ object SequentialTagIndexer {
-                index(value).map(_ => off)
+                retry(() => index(value), retries, backoff)
+                  .recoverWith { case _ => failureLog.storeEvent(off, value) }
+                  .map(_ => off)
               case None =>
                 log.debug(s"Event not compatible with type '${T.describe}, skipping...'")
                 Future.successful(off)
             }
         }
         .mapAsync(1)(offset => projection.storeLatestOffset(offset))
   }
+  private def lookupRetriesConfig(implicit as: ActorSystem): (Int, RetryStrategy) = {
+    val config  = as.settings.config.getConfig("indexing.retries")
+    val retries = config.getInt("times")
+    val backoff = Backoff(Duration(config.getDuration("max", SECONDS), SECONDS), config.getDouble("random-factor"))
+    (retries, backoff)
+  }
 
   /**
     * Generic tag indexer that uses the specified resumable projection to iterate over the collection of events selected
     * via the specified tag and apply the argument indexing function.  It starts as a singleton actor in a
     * clustered deployment.  If the event type is not compatible with the events deserialized from the persistence store
     * the events are skipped.
     *
-    * @param init         an initialization function that is run before the indexer is (re)started
-    * @param index        the indexing function
-    * @param projectionId the id of the resumable projection to use
-    * @param pluginId     the persistence query plugin id
-    * @param tag          the tag to use while selecting the events from the store
-    * @param name         the name of this indexer
-    * @param as           an implicitly available actor system
-    * @param T            a Typeable instance for the event type T
+    * @param init     an initialization function that is run before the indexer is (re)started
+    * @param index    the indexing function
+    * @param id       the id of the resumable projection and indexing log to use
+    * @param pluginId the persistence query plugin id
+    * @param tag      the tag to use while selecting the events from the store
+    * @param name     the name of this indexer
+    * @param as       an implicitly available actor system
+    * @param T        a Typeable instance for the event type T
+    * @param E        an implicitly available [[io.circe.Encoder]] for T
     * @tparam T the event type
     */
   // $COVERAGE-OFF$
   final def start[T](init: () => Future[Unit],
                      index: T => Future[Unit],
-                     projectionId: String,
+                     id: String,
                      pluginId: String,
                      tag: String,
-                     name: String)(implicit as: ActorSystem, T: Typeable[T]): ActorRef =
-    SingletonStreamCoordinator.start(initialize(init, pluginId), source(index, projectionId, pluginId, tag), name)
+                     name: String)(implicit as: ActorSystem, T: Typeable[T], E: io.circe.Encoder[T]): ActorRef =
```

Comment:
Any reason the encoder is fully qualified?

Reply:
We need the implicit Encoder[T] because `IndexFailuresLog.storeEvent` needs it.